### PR TITLE
SED-2147 analyse artefactjs migrate simple directive

### PR DESCRIPTION
**Pull request by dvladir**

### CODE REVIEW

**Review by neogucky — APPROVED** [review]

**Review by bojidaryovchev — APPROVED** [review]

File: projects/step-core/src/lib/modules/dynamic-forms/components/dynamic-textfield/dynamic-textfield.component.html [hunk 1]
```diff
@@ -14,14 +14,26 @@
 <ng-template #constantValueTemplate>
   <step-form-field [class.ng-invalid]="!!_ngControl.invalid" [class.with-tooltip]="tooltip">
     <step-label [attr.show-required-marker]="showRequiredAsterisk || null" *ngIf="label">{{ label }}</step-label>
-    <input
-      [type]="isNumber ? 'number' : 'text'"
-      [required]="showRequiredAsterisk"
```
Comment (bojidaryovchev) [inline projects/step-core/src/lib/modules/dynamic-forms/components/dynamic-textfield/dynamic-textfield.component.html:17]: this part is no longer present it seems, do we need it?

File: projects/step-core/src/lib/modules/dynamic-forms/components/dynamic-textfield/dynamic-textfield.component.ts [hunk 2]
```diff
@@ -54,6 +54,13 @@ export class DynamicTextfieldComponent implements ControlValueAccessor {
     this.emitChanges();
   }
 
+  preventNumberInvalidChars(event: KeyboardEvent): void {
+    const numberInputInvalidCharts = ['+', '-', 'e'];
```
Comment (bojidaryovchev) [inline projects/step-core/src/lib/modules/dynamic-forms/components/dynamic-textfield/dynamic-textfield.component.ts:58]: ```numberInputInvalidCharts``` => ```numberInputInvalidChars```

Comment (bojidaryovchev) [reply projects/step-core/src/lib/modules/dynamic-forms/components/dynamic-textfield/dynamic-textfield.component.ts:58]: also maybe worth for this to be extracted in a directive that would be applied on an input and provided with forbidden symbols

**Review by bojidaryovchev — APPROVED** [review]

File: projects/step-core/src/lib/modules/dynamic-forms/components/dynamic-textfield/dynamic-textfield.component.html (hunk 1 above)
Comment (dvladir) [reply projects/step-core/src/lib/modules/dynamic-forms/components/dynamic-textfield/dynamic-textfield.component.html:17]: Input validity should depends on validator that is assigned to formControl.
It's not straightforward, that we have a component that implements `ControlValueAccessor` plus it has it's own validation internal logic.
Plus the attributes name `showRequiredAsterisk` informs that it's more about visual view.

Comment (bojidaryovchev) [reply projects/step-core/src/lib/modules/dynamic-forms/components/dynamic-textfield/dynamic-textfield.component.html:17]: yea I guess at the time I put this we didnt have the ```[attr.show-required-marker]="showRequiredAsterisk || null"``` part so I put the required part in order for the asterisk to be shown

**Pull request merged**